scripts/create_irreducible_polys_database.py

```diff
@@ -0,0 +1,111 @@
+"""
+A script to create a database of irreducible polynomials
+
+Sources:
+ - Gadiel Seroussi. Table of Low-Weight Binary Irreducible Polynomials (1998): https://www.hpl.hp.com/techreports/98/HPL-98-135.html
+
+"""
+from __future__ import annotations
+
+import os
+import sqlite3
+from pathlib import Path
+
+import requests
+import hashlib
+from pdfminer.high_level import extract_text
+
+
+def main():
+    """
+    The main routine to create a database of irreducible polynomials
+    """
+
+    database_file = Path(__file__).parent.parent / "src" / "galois" / "_databases" / "irreducible_polys.db"
+    conn, cursor = create_database(database_file)
+
+    _add_hpl_1998(conn, cursor)
+
+    conn.close()
+
+
+def create_database(file: Path) -> tuple[sqlite3.Connection, sqlite3.Cursor]:
+    """
+    Deletes the old database, makes a new one, and returns the database connection.
+    """
+    if file.exists():
+        os.remove(file)
+
+    conn = sqlite3.connect(file)
+    cursor = conn.cursor()
+    create_table(conn, cursor)
+
+    return conn, cursor
+
+
+def create_table(conn: sqlite3.Connection, cursor: sqlite3.Cursor):
+    """
+    Creates an empty 'polys' table.
+    """
+    cursor.execute(
+        """
+        CREATE TABLE polys (
+            characteristic INTEGER NOT NULL,
+            degree INTEGER NOT NULL,
+            nonzero_degrees TEXT NOT NULL,
+            nonzero_coeffs TEXT NOT NULL,
+            PRIMARY KEY (characteristic, degree)
+        )
+        """
+    )
+    conn.commit()
+
+
+def add_to_database(cursor: sqlite3.Cursor, characteristic: str, degree: str, nonzero_degrees: str, nonzero_coeffs: str):
+    """
+    Adds the given irreducible polynomial to the database.
+    """
+    cursor.execute(
+        """
+        INSERT INTO polys (characteristic, degree, nonzero_degrees, nonzero_coeffs)
+        VALUES (?,?,?,?)
+        """,
+        (int(characteristic), int(degree), nonzero_degrees, nonzero_coeffs),
+    )
+
+
+def _add_hpl_1998(conn, cursor):
+    """
+    Add Gadiel Seroussi's table to the database.
+    GF(2^m) for 2 <= m <= 10_000
+    """
+    url = "https://www.hpl.hp.com/techreports/98/HPL-98-135.pdf"
+    # There is an issue with the SSL certificate using CURL_CA_BUNDLE
+    # We don't validate https, but we do check the PDF's checksum
+    pdf = requests.get(url, stream=True, verify=False).content
+    sha256 = hashlib.sha256()
+    sha256.update(pdf)
+    assert sha256.hexdigest() == "78f02d84a0957ad261c53a0d1107adb2ff9d72f52ba5e10ea77eaa8cf766a0ee"
```

Comment:
What do you think of this solution? I prefer this than having to download the certificate chain and passing it to `requests.get()`. I don't think it's critical to use https here, just that the PDF is the correct one.

Reply:
I like this solution!